static/js/apps/visualization/vis_type_configs/map_config.tsx

```diff
@@ -57,12 +58,17 @@ function getFacetSelector(appContext: AppContextType): JSX.Element {
       },
     ];
   });
-  const onSvFacetIdUpdated = (svFacetId: Record<string, string>) => {
+  const onSvFacetIdUpdated = (
+    svFacetId: Record<string, string>,
+    metadataMap: Record<string, StatMetadata>
+  ) => {
     if (svFacetId[statVar.dcid] === statVar.facetId) {
       return;
     }
     const newStatVars = _.cloneDeep(appContext.statVars);
+    const facetId = svFacetId[newStatVars[0].dcid];
     newStatVars[0].facetId = svFacetId[newStatVars[0].dcid];
+    newStatVars[0].facetInfo = metadataMap[facetId];
```

Comment:
Can newStatVars ever be empty?

Reply:
newStatVars should never be empty because the facet selector is only shown when there is a map chart and there can only be a map chart if there is at least one statvar.. but added a check just in case.